Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -384,13 +384,68 @@ extension SwiftLanguageServer {
           }
           let isInsertTextSnippet = clientSupportsSnippets && text != insertText
 
+          let textEdit: TextEdit?
+          if let text = text {
+            let utf8CodeUnitsToErase: Int = value[self.keys.num_bytes_to_erase] ?? 0
+
+            // Compute the TextEdit
+            if utf8CodeUnitsToErase == 0 {
+              // Nothing to delete. Fast path and avoid UTF-8/UTF-16 conversions
+              textEdit = TextEdit(range: completionPos..<req.params.position, newText: text)
+            } else if utf8CodeUnitsToErase == 1 {
```

Comment:
This code is getting pretty gnarly, could we factor this into its own function to compute the edit?  Also, can we rearrange it to

1. compute the start location for the range
2. fallback for the multiline case

so that the logic can be shared between the count == 1 case and the general case? I think that would make it clearer what the overall flow is.